medicine-service/src/main/java/cz/fi/muni/pv217/narcos/medicine/resource/MedicineResource.java

```diff
@@ -58,25 +59,56 @@ public Response getMedicineById(@PathParam("id") Long id) {
     @RolesAllowed({"Admin", "User"})
     @Produces(MediaType.APPLICATION_JSON)
     public Response getMedicineByName(@PathParam("name") String name) {
-        LOG.info(String.format("Starting procedure get user with name '%s'.", name));
+        LOG.info(String.format("Starting procedure get medicine with name '%s'.", name));
         Medicine medicine = medicineRepository.findByName(name);
 
         if (medicine == null) {
-            LOG.error(String.format("User with name '%s' doesn't exists.", name));
+            LOG.error(String.format("Medicine with name '%s' doesn't exists.", name));
             throw new WebApplicationException(403);
         }
 
         return Response.ok(medicineToMedicineDTO(medicine)).build();
     }
 
+    @POST
+    @Path("{id}")
```

Comment:
Create endpoint should not take id as path parameter as user does not know what id will be used and id is not used in code. Please remove it from there.